server/src/main/java/org/jboss/as/server/deployment/DeploymentUnitPhaseService.java

```diff
@@ -277,4 +281,67 @@ private static boolean shouldRun(final DeploymentUnit unit, final RegisteredDepl
         }
         return !shouldNotRun.contains(deployer.getSubsystemName());
     }
+
+    static class DeploymentUnitServiceTarget extends DelegatingServiceTarget implements ServiceTarget {
+        private final CapabilityServiceSupport support;
+
+        DeploymentUnitServiceTarget(org.jboss.msc.service.ServiceTarget target, CapabilityServiceSupport support) {
+            super(target);
+            this.support = support;
+        }
+
+        @Override
+        public ServiceTarget addListener(LifecycleListener listener) {
```

Comment:
Why returning ServiceTarget instead of DeploymentUnitServiceTarget? Is it intentional?

Reply:
Because there is nothing to be gained by overloading the return type in the implementation, which is not visible to consumers.

Reply:
Makes sense. Thanks for clarification!